Framework/DataHandling/test/ISISJournalTest.h

```diff
@@ -0,0 +1,216 @@
+                                      {"run_number", "12347"},
+                                      {"title", "Experiment 1 run 2 th=2.3"}},
+                                     {{"name", "INTER00052345"},
+                                      {"run_number", "52345"},
+                                      {"title", "Experiment 2 trans 2 3 45"}}};
+    TS_ASSERT_EQUALS(results, expected);
+  }
+
+  void test_getCycleList_with_empty_file_throws() {
+    TS_ASSERT_THROWS(getCycleListFromFile(emptyFile), std::runtime_error);
+  }
+
+  void test_getCycleList_with_bad_xml_throws() {
+    TS_ASSERT_THROWS(getCycleListFromFile(badFile), std::runtime_error);
+  }
+
+  void test_getCycleList_with_empty_xml_file_returns_empty_results() {
+    auto results = getCycleListFromFile(emptyIndexFile);
+    TS_ASSERT_EQUALS(results, std::vector<std::string>{});
+  }
+
+  void test_getCycleList_throws_when_invalid_element_names() {
+    TS_ASSERT_THROWS(getCycleListFromFile(invalidIndexFile),
+                     std::invalid_argument);
+  }
+
+  void test_getCycleList_returns_all_valid_cycles() {
+    auto results = getCycleListFromFile(indexFile);
+    auto const expected = std::vector<std::string>{"17_1", "17_2", "18_1",
+                                                   "19_1", "19_2", "19_3"};
+    TS_ASSERT_EQUALS(results, expected);
+  }
+
+private:
+  ISISJournalTags defaultTags() {
+    return ISISJournalTags{"run_number", "title"};
+  }
+
+  ISISJournalTags emptyTags() { return ISISJournalTags{}; }
+
+  ISISJournalFilters defaultFilters() {
+    return ISISJournalFilters{{"experiment_identifier", "1112233"}};
+  }
+
+  ISISJournalFilters multipleFilters() {
+    return ISISJournalFilters{{"experiment_identifier", "1112233"},
+                              {"number_detectors", "5"}};
+  }
+
+  ISISJournalFilters emptyFilters() { return ISISJournalFilters{}; }
+
+  static constexpr const char *emptyFile = "";
+
+  static constexpr const char *badFile = "<NXroot";
+
+  static constexpr const char *emptyJournalFile = "\
+<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
+      <NXroot NeXus_version=\"4.3.0\" XML_version=\"mxml\"></NXroot>";
+
+  static constexpr const char *emptyIndexFile = "<journal></journal>";
+
+  static constexpr const char *invalidIndexFile =
+      "<journal><badtag/></journal>";
+
+  static constexpr const char *indexFile = "\
+<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\
+<?xml-stylesheet type=\"text/xsl\" href=\"journal.xsl\" ?>\
+<!-- DO NOT EDIT - recreated automatically by ICP at each END of run -->\
+<journal>\
+  <file name=\"journal.xml\" />\
+  <file name=\"journal_17_1.xml\" />\
+  <file name=\"journal_17_2.xml\" />\
+  <file name=\"journal_18_1.xml\" />\
+  <file name=\"journal_19_1.xml\" />\
+  <file name=\"journal_19_2.xml\" />\
+  <file name=\"journal_19_3.xml\" />\
+</journal>\
+";
+
+  static constexpr const char *journalFile = "\
+<?xml version=\"1.0\" encoding=\"UTF-8\"?>\
```

Comment:
I'm not keen on embedding larger XML docs (such as this case) into the test file, however I also understand loading in data at test time is a pain too

So this one is your call, personally I'd prefer to see this example (the rest are fine) moved into a test file

Reply:
It's probably larger than it needs to be tbh. I'll first off look at reducing the size.